detectable/src/main/java/com/synopsys/integration/detectable/detectables/pnpm/lockfile/process/PnpmYamlTransformer.java

```diff
@@ -114,18 +126,41 @@ private String convertRawEntryToPackageId(Map.Entry<String, String> entry, PnpmL
         return String.format("/%s/%s", name, version);
     }
 
-    private NameVersion parseNameVersionFromId(String id) {
+    private Optional<NameVersion> parseNameVersionFromId(String id) {
         // ids follow format: /name/version, where name often contains slashes
-        int indexOfLastSlash = id.lastIndexOf("/");
-        String name = id.substring(1, indexOfLastSlash);
-        String version = id.substring(indexOfLastSlash + 1);
+        try {
```

Comment:
I see you added a test for file dependencies, but I don't understand how this code handles the  `file:` prefix.

Reply:
for file dependencies, we cannot reliably parse the name of the file itself for name/version info so in buildDependencyFromPackageEntry we look to see if there is a name and version specified in the package entry (which, from what I've seen, is the protocol for file dependencies but not other types of dependencies).  if we find them, we use those to create the external id, and if we don't then we'll hit the debug message in parseNameVersionFromId and return an empty Optional